src/generator/writers/writeFunction.ts

```diff
@@ -557,6 +557,33 @@ export function writeStatement(
     throw Error("Unknown statement kind");
 }
 
+const rewriteWithIfNot = (expr: Ast.Expression): [Ast.Expression, string] => {
+    if (expr.kind === "op_unary" && expr.op === "!") {
+        // `if (~ cond)` => `ifnot (cond)`
+        return [expr.operand, "ifnot"];
+    }
+    if (
+        expr.kind === "op_binary" &&
+        expr.op === "==" &&
+        expr.right.kind === "number" &&
+        expr.right.value === 0n
```

Comment:
- We should also take care of the symmetrical `0 == a` case.
- The optimization should work for any compile-time expression that evaluates to `0`, see how it is implemented in `writeExpressionInCondition`

Reply:
Done